source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py

```diff
@@ -157,12 +160,55 @@ def _do_request(self):
-        places_isochrone = [p for p in proximities_by_crowfly if p.uri not in all_free_access]
+        access_points_map = defaultdict(list)
 
-        result = {}
+        if self._mode == FallbackModes.car.name or self._request['_access_points'] is False:
+            # if a place is freely accessible, there is no need to compute it's access duration in isochrone
+            places_isochrone = [p for p in proximities_by_crowfly if p.uri not in all_free_access]
+        else:
+            places_isochrone = []
+            for p in proximities_by_crowfly:
+                # if a place is freely accessible, there is no need to compute it's access duration in isochrone
+
+                if p.uri in all_free_access:
+                    continue
+                # what we are looking to compute, is not the stop_point, but the entrance and exit of a stop_point
+                # if any of them are existent
+                if not p.stop_point.access_points:
+                    places_isochrone.append(p)
+
+                if self._direct_path_type == StreetNetworkPathType.BEGINNING_FALLBACK:
+                    for ap in (ap for ap in p.stop_point.access_points if ap.is_entrance):
+                        if ap.uri not in access_points_map:
+                            places_isochrone.append(
+                                type_pb2.PtObject(
+                                    name=ap.name,
+                                    uri=ap.uri,
+                                    embedded_type=type_pb2.ACCESS_POINT,
+                                    access_point=ap,
+                                )
+                            )
+                        access_points_map[ap.uri].append((p.stop_point.uri, ap.length, ap.traversal_time))
+
+                if self._direct_path_type == StreetNetworkPathType.ENDING_FALLBACK:
+                    for ap in (ap for ap in p.stop_point.access_points if ap.is_exit):
+                        if ap.uri not in access_points_map:
+                            places_isochrone.append(
+                                type_pb2.PtObject(
+                                    name=ap.name,
+                                    uri=ap.uri,
+                                    embedded_type=type_pb2.ACCESS_POINT,
+                                    access_point=ap,
+                                )
+                            )
+                        access_points_map[ap.uri].append((p.stop_point.uri, ap.length, ap.traversal_time))
+
+        result = defaultdict(lambda: DurationElement(float('inf'), None, None, 0, None))
         # Since we have already places that have free access, we add them into the result
-        [result.update({uri: DurationElement(0, response_pb2.reached, None, 0)}) for uri in all_free_access]
+        [
+            result.update({uri: DurationElement(0, response_pb2.reached, None, 0, None)})
+            for uri in all_free_access
+        ]
```

Comment:
This creates a list but do nothing with it ??

Reply:
it's a pythonic way to do for_each :)